### PR TITLE
Font awesome automatic updating and refactor

### PR DESCRIPTION
**Pull request by bassettsj**
Built a nifty grunt task to convert the yaml from their jekyll site into a json blog that we can use to return all the icons.

### CODE REVIEW

**Comment by bassettsj** [comment]
resolves #45

File: includes/icons/fontawesome.inc [hunk 1]
```diff
@@ -1,465 +1,34 @@
...
-    'fa-ambulance' => 'ambulance',
-    'fa-h-square' => 'h-square',
-    'fa-hospital-o' => 'hospital-o',
-    'fa-medkit' => 'medkit',
-    'fa-plus-square' => 'plus-square',
-    'fa-stethoscope' => 'stethoscope',
-    'fa-user-md' => 'user-md',
-    'fa-wheelchair' => 'wheelchair'
-  );
+  $icons = &drupal_static(__FUNCTION__);
```
Comment (bassettsj) [inline includes/icons/fontawesome.inc]: @pirog & @reynoldsalec is this caching overkill or does it make sense? I wasn't sure how expensive it is to read a file decode it and iterate through it.

**Comment by pirog** [comment]
@bassettsj i would look into this a bit more. i _think_ APC automatically caches all returns via file_get_contents so this might actually be slower although i doubt by much if at all. We also might not want to assume APC is installed so this would make more sense. 

Another key question here is how many times is this function going to be called for a given page load?

**Comment by bassettsj** [comment]
I don't think it is called often at all. I think it is just called when icon api is initiated and then the hooks results are saved. Would have to look at icon_api source.

**Comment by pirog** [comment]
Yeah if its not called a ton im not sure caching the function return will make that big a difference. 
@reynoldsalec, do you have anything to offer here?

**Comment by bassettsj** [comment]
@pirog seems like pre-optimizing to me...

So glad we're getting onto the code review train, it will be good for the whole team to start doing this more often. :rainbow:

**Comment by pirog** [comment]
@bassettsj so am i.... it will be great to get kalabox in a place where we can start doing this sort of thing a lot easier for our project work as well.

**Comment by reynoldsalec** [comment]
This looks exactly how I would do it; if performance becomes an issue we can always add some conditional logic to look for APC and not bother with the cache_set if it is around, but I'm guessing this is all negligible.

**Comment by bassettsj** [comment]
So should I just simplify the the function to something like this? 

``` php
$icons = array();
  if (file_exists(__DIR__ . '/fontawesome.json')) {
  $file = file_get_contents(__DIR__ . '/fontawesome.json');
      $raw_icons = json_decode($file, TRUE);
       foreach ($raw_icons['icons'] as $delta => $icon) {
          $icons[$icon['id']] = $icon['name'];
        }
  }

```

**Pull request merged**